kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/storage/cloud/CloudClientFactory.java

```diff
@@ -0,0 +1,33 @@
+package org.kie.server.services.impl.storage.cloud;
+
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.kie.server.api.KieServerConstants;
+
+public interface CloudClientFactory {
```

Comment:
Is there any reason to have this as interface?
I think it could be done as an utility class.

Reply:
The intention is to allow implementation of different ways of creating OpenShift/K8S client. For example, a client is retrieved from Mock OpenShift server at unit test class.

Reply:
No problem with functionality point of view.
I just don't like it to be implemented as an interface. It is used for example in OpenShiftStartupStrategy. So now OpenShiftStartupStrategy has a public method createAdaptiveClient() which IMHO doesn't have any relation to its own functionality.
I think it is better to move this functionality to separate util or factory class, dedicated to provide OpenShift/K8S client.

Reply:
Points are taken. Made the change. The CloudClientFactory interface is not implemented by OpenShiftStartupStrategy anymore.